core/src/main/java/com/linecorp/armeria/client/HttpClientDelegate.java

```diff
@@ -247,9 +282,14 @@ private static void logSession(ClientRequestContext ctx, @Nullable PooledChannel
         }
     }
 
-    private static HttpResponse earlyFailedResponse(Throwable t, ClientRequestContext ctx) {
+    private static Throwable earlyFailed(Throwable t, ClientRequestContext ctx) {
```

Comment:
nit: How about taking an `HttpResponse` to close it here?
```suggestion
    private static HttpResponse earlyFailedResponse(Throwable t, ClientRequestContext ctx, HttpResponse res) {
        final UnprocessedRequestException cause = UnprocessedRequestException.of(t);
        ctx.cancel(cause);
        res.close(cause);
        return res;
```